### PR TITLE
Start expansion service containers before the controller container

### PR DESCRIPTION
**Pull request by chamikaramj**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @ahmedabu98

**Comment by ahmedabu98** [comment]
If we're starting both containers at the same time, we should probably have different port numbers so they don't conflict: 
https://github.com/apache/beam/blob/0c55d382988b510da37fa1c3d193ae949a07dfe9/sdks/java/transform-service/docker-compose/docker-compose.yml#L24-L35

File: sdks/java/transform-service/docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -18,6 +18,9 @@ services:
     command: -port 5001
     ports:
       - "${TRANSFORM_SERVICE_PORT}:5001"
+    depends_on:
+      - expansion-service-1
+      - expansion-service-2
```
Comment (ahmedabu98) [inline sdks/java/transform-service/docker-compose/docker-compose.yml:23]: I'm not too familiar with how this works. Does this mean both expansion services will be built and run whenever we use the transform service?

Comment (chamikaramj) [reply sdks/java/transform-service/docker-compose/docker-compose.yml:23]: This is a docker compose feature. It allows specifying a dependency order between services (containers) of the Docker Compose service.

https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on

**Comment by chamikaramj** [comment]
"5001" is the internal port number of the containers (not the external port) so there's no port conflict here.

File: sdks/java/transform-service/docker-compose/docker-compose.yml (hunk 1 above)
Comment (ahmedabu98) [reply sdks/java/transform-service/docker-compose/docker-compose.yml:23]: Ok I see. Maybe later we should consider specifying the services needed for transform-service. Most use-cases won't need both java and python expansion services. Not blocking though, this LGTM

**Review by ahmedabu98 — APPROVED** [review]

File: sdks/java/transform-service/docker-compose/docker-compose.yml (hunk 1 above)
Comment (chamikaramj) [reply sdks/java/transform-service/docker-compose/docker-compose.yml:23]: > Most use-cases won't need both java and python expansion services

I'm hoping to use this as a generalized service that support all transforms. The main cost will be container download and storage. Once downloaded the expansion service containers can be started very quickly. Also, hoping to push changes to reduce size of the Java container in the future by removing the duplications due to using multiple shaded jars.

**Comment by chamikaramj** [comment]
Thanks.

**Comment by chamikaramj** [comment]
(test failures are unrelated)

**Pull request merged**